// containers/books/index.js
import React, {Component} from 'react'
import {connect} from 'react-redux'
import * as R from 'ramda'

import {getBooksById} from 'selectors'
import {fetchBooksById} from 'actions'

class Books extends Component {
    componentDidMount () {
        this.props.fetchBooksById(this.props.params.id)
    }

    renderFields () {
        const {books} = this.props;
        const columnFields = R.compose (
            R.toPairs,
            R.pick([
                'pages' 
            ])
        )(books)

        return columnFields.map(([key, value]) => (
            <div className='column' key={key}>
                <div className='ab-details-title'>
                    <p>{key}</p>
                </div>
                <div className='ab-details-info'>
                    {value}
                </div>
            </div>
        ))
    }

    renderContent () {
        const {books} = this.props

        return (
            <div className='thumbnail'>
                <div className='row'>
                    <div className='col-md-6'>
                        <img
                            className='img-thumbnail'
                            src={books.image}
                            alt={books.name}
                        />
                    </div>
                    <div className='col-md-6'>
                        {this.renderFields()}
                    </div>
                </div>
                <div className='caption-full'>
                    <h4 className='pull-right'>${books.price}</h4>
                    <h4>{books.name}</h4>
                    <p>{books.description}</p>
                </div>
            </div>
        )
    }

    renderSidebar () {
        return (
            <div>Sidebar</div>
        )
    }

    render () {
        const {books} = this.props
        return (
            <div className='view-container'>
                <div className='container'>
                    <div className='row'>
                        <div className='col-md-9'>
                            {books && this.renderContent()}
                        </div>
                        <div className='col-md-3'>
                            {books && this.renderSidebar()}
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}

const mapStateToProps = state => {
    return {
        books: getBooksById(state, state.booksPage.id)
    }
}

const mapDispatchToProps = {
    fetchBooksById
}

export default connect(mapStateToProps, mapDispatchToProps)(Books)
