// api/mockBook.js
export default [
    {
        id: "1",
        categoryId: "1",
        name: "War and Peace",
        description: "lorem asfa asf saf ",
        price: 45,
        image: '/uploads/war_and_pleace.jpg',
        pages: 1000
    },

    {
        id: "2",
        categoryId: "2",
        name: "The Creat Gatsby",
        description: "lorem asfa asf saf ",
        price: 35,
        image: '/uploads/gatsby.png',
        pages: 1020,
    },

    {
        id: "3",
        categoryId: "2",
        name: "Tender Is the Night",
        description: "lorem asfa asf saf ",
        price: 20,
        image: '/uploads/Night.jpg',
        pages: 600
    },

    {
        id: "4",
        categoryId: "3",
        name: "Kammerater",
        description: "lorem asfa asf saf ",
        price: 80,
        image: '/uploads/kammerater.png',
        pages: 1234
    },

    {
        id: "5",
        categoryId: "4",
        name: "O Alquimista",
        description: "lorem asfa asf saf ",
        price: 50,
        image: '/uploads/Alquimista.jpg',
        pages: 700
    },

    {
        id: "6",
        categoryId: "5",
        name: "The Da Vinci Code",
        description: "lorem asfa asf saf ",
        price: 45,
        image: '/uploads/code.jpg',
        pages: 900
    }

]
